core/shared/src/main/scala/fs2/internal/Algebra.scala

```diff
@@ -49,94 +64,50 @@ private[fs2] object Algebra {
 
   private[this] final case class GetScope[F[_]]() extends AlgEffect[F, CompileScope[F]]
 
-  def output[F[_], O](values: Chunk[O]): FreeC[Algebra[F, O, ?], Unit] =
-    FreeC.Eval[Algebra[F, O, ?], Unit](Output(values))
+  def output[F[_], O](values: Chunk[O]): FreeC[F, O, Unit] = Output(values)
 
-  def output1[F[_], O](value: O): FreeC[Algebra[F, O, ?], Unit] =
-    output(Chunk.singleton(value))
+  def output1[F[_], O](value: O): FreeC[F, O, Unit] = Output(Chunk.singleton(value))
 
-  def eval[F[_], O, R](value: F[R]): FreeC[Algebra[F, O, ?], R] =
-    FreeC.Eval[Algebra[F, O, ?], R](Eval(value))
+  def eval[F[_], R](value: F[R]): FreeC[F, INothing, R] = Eval(value)
 
   def acquire[F[_], O, R](
       resource: F[R],
       release: (R, ExitCase[Throwable]) => F[Unit]
-  ): FreeC[Algebra[F, O, ?], R] =
-    FreeC.Eval[Algebra[F, O, ?], R](Acquire(resource, release))
-
-  def mapOutput[F[_], A, B](fun: A => B): Algebra[F, A, ?] ~> Algebra[F, B, ?] =
-    new (Algebra[F, A, ?] ~> Algebra[F, B, ?]) {
-      def apply[R](alg: Algebra[F, A, R]): Algebra[F, B, R] = alg match {
-        case o: Output[F, A] => Output[F, B](o.values.map(fun))
-        case _               => alg.asInstanceOf[Algebra[F, B, R]]
-      }
-    }
+  ): FreeC[F, O, R] =
+    Acquire(resource, release)
```

Comment:
@mpilquist For now, I have only unfolded/removed the factory methods that were private to this object. However, since the whole `Algebra` is private to `fs2`, would it make sense to open  the subclasses and remove this method, as well as `eval` and `output` above?